### PR TITLE
Allow running `python -m ice.server`

### PR DESCRIPTION
**Pull request by alexmojaki**

### CODE REVIEW

File: ice/server.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from uvicorn import main
```
Comment (lslunis) [inline ice/server.py:1]: Why main instead of run?

https://www.uvicorn.org/deployment/#running-programmatically

File: ice/server.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+params_by_name = {p.name: p for p in main.params}
```
Comment (lslunis) [inline ice/server.py:3]: How does this variable get used?

File: ice/server.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+params_by_name["app"].default = "ice.routes.app:app"
+params_by_name["port"].default = 8935
```
Comment (lslunis) [inline ice/server.py:5]: It'd be nice to have reload=True when installing ice in editable mode. Can we detect that?

Alternatively, maybe for now we should always run with reload=True

File: ice/server.py (hunk 1 above)
Comment (alexmojaki) [reply ice/server.py:1]: This allows reusing all the `click` options, e.g. `python -m ice.server --port ... --host ...`

File: ice/server.py (hunk 2 above)
Comment (alexmojaki) [reply ice/server.py:3]: It's not meant to be used externally.

Comment (lslunis) [reply ice/server.py:3]: Never mind, just noticed the existing defaults are getting mutated

File: ice/server.py (hunk 3 above)
Comment (alexmojaki) [reply ice/server.py:5]: I think for the sake of exernal package users it's best not to touch the default of reload. Internally it's easy to add `--reload` in development.

**Review by lslunis — APPROVED** [review]

**Pull request merged**